### PR TITLE
Switch to xUnit.net 2 MSBuild runner

### PR DESCRIPTION
**Pull request by TeaDrivenDev**
This switches the build process to using the MSBuild runner for xUnit.net 2. This is necessary to be able to run xUnit 2 tests, which will be needed for testing the xUnit 2 glue library. The xUnit 2 runner also runs xUnit 1 tests, so none of the existing tests will need to be changed.

The changes are as follows:
- `NuGet.exe` has been updated to the current 2.8 release, because the xUnit 2 runner package required it.
- The xUnit 1 runner has been removed from the `Lib` directory and replaced by the xUnit 2 MSBuild runner installed via NuGet in the `Packages` directory.
- The MSBuild file has been updated to use the xUnit 2 runner for the existing xUnit tests.
- The `DeleteNUnitTestResultFile` build target has been renamed to account for the fact that there will also be xUnit test result files to delete now.
- xUnit test discovery is now rule based and controlled from the MSBuild file. Therefore `All.xunit` has been deleted.

**Note:** This is the first of nearly 20 pull requests to enable xUnit 2 integration on par with the current state for xUnit 1. I have most of the work already prepared, but due to the way Git merging works will only be able to have pull requests open at the same time that do not depend on each other.

### CODE REVIEW

File: BuildRelease.msbuild [hunk 1]
```diff
@@ -95,12 +95,30 @@
...
-        <xunitproject ProjectFile="Src\All.xunit" />
+        <PropertyGroup>
+          <Configuration Condition="'$(Configuration)' == ''">Release</Configuration>
+          <ParallelizeTests Condition="'$(ParallelizeTests)' == ''">true</ParallelizeTests>
+          <MaxParallelThreads Condition="'$(MaxParallelThreads)' == ''">0</MaxParallelThreads>
+        </PropertyGroup>
+
+        <ItemGroup>
+          <Xunit1TestAssemblies
+              Include="Src\*UnitTest\bin\$(Configuration)\*UnitTest.dll"
```
Comment (moodmosaic) [inline BuildRelease.msbuild]: How about using a pattern like `*Test.dll`? Will the proposed pattern (`*UnitTest.dll`) match

```
AutoFixtureDocumentationTest\bin\Release\Ploeh.AutoFixtureDocumentationTest.dll
```
?

**Comment by TeaDrivenDev** [comment]
You are of course correct; it didn't. I have updated the pattern to be more inclusive (and also used the opportunity to simplify the NUnit test exclusion pattern).

**Comment by moodmosaic** [comment]
Nice! It looks like the v2 runner reports the grand total. Here's an excerpt from the build output:

```
=== TEST EXECUTION SUMMARY ===
Ploeh.AutoFixture.AutoFakeItEasy.UnitTest   Total:   68, Errors: 0, Failed: 0, Skipped: 0, Time: 0.828s
Ploeh.AutoFixture.AutoFoq.UnitTest          Total:   23, Errors: 0, Failed: 0, Skipped: 0, Time: 0.590s
Ploeh.AutoFixture.AutoMoq.UnitTest          Total:  169, Errors: 0, Failed: 0, Skipped: 0, Time: 0.751s
Ploeh.AutoFixture.AutoNSubstitute.UnitTest  Total:  129, Errors: 0, Failed: 0, Skipped: 0, Time: 0.735s
Ploeh.AutoFixture.AutoRhinoMock.UnitTest    Total:   63, Errors: 0, Failed: 0, Skipped: 0, Time: 0.616s
Ploeh.AutoFixture.Idioms.FsCheckUnitTest    Total:   27, Errors: 0, Failed: 0, Skipped: 0, Time: 0.750s
Ploeh.AutoFixture.IdiomsUnitTest            Total:  579, Errors: 0, Failed: 0, Skipped: 0, Time: 0.993s
Ploeh.AutoFixture.Xunit.UnitTest            Total:   92, Errors: 0, Failed: 0, Skipped: 0, Time: 0.199s
Ploeh.AutoFixtureDocumentationTest          Total:   96, Errors: 0, Failed: 0, Skipped: 0, Time: 0.453s
Ploeh.AutoFixtureUnitTest                   Total: 3567, Errors: 0, Failed: 0, Skipped: 0, Time: 2.355s
Ploeh.SemanticComparison.UnitTest           Total:  247, Errors: 0, Failed: 0, Skipped: 0, Time: 0.379s
                                                  ----          -          -           -        ------
                                     GRAND TOTAL: 5060          0          0           0        8.649s (17.621s)
```

**Comment by moodmosaic** [comment]
It's looking good to me :+1:

**Comment by ploeh** [comment]
Thank you for your contribution! Since this is a test-only update, no published version is associated with it.
